WordPress/gutenberg
# Add a filter to `build_query_vars_from_query_block`

## What?
This PR adds a filter to the `build_query_vars_from_query_block` which is used to parse a `WP_Query` and pass it down the Query Loop block context. 

## Why?
This allows Query Loop block variations to hook into the query and possibly inject custom query parameters, enabling developers to support all kinds of custom queries and create powerful variations on the Query Loop block.

For an example on how this can be used to achieve this goal, see this PR: https://github.com/woocommerce/woocommerce-blocks/pull/6812 (specifically, this filter is used [here](https://github.com/woocommerce/woocommerce-blocks/pull/6812/files#diff-efb947e345af5e501dfaa2980b055b3b31c308b31e8db23049b6b48fecee33d8R39-R52)).

Paging @gziolo and @ntsekouras.

## How?

The minimal change to enable this is to wrap the return value of the function in a filter.

## Testing Instructions

1. Add a callback to your PHP code that hooks to this filter, e.g.

```
add_filter(
  'gutenberg_build_query_vars_from_query_block',
  function ( $query ) {
    $query[ 'offset' ] = 1;

    return $query;
 },
 10,
 1
);
```

2. Add multiple Query Loop blocks to your pages with different settings.
3. Save the page and go to the front-end.
4. Notice that all the blocks will render with your modified query (e.g. start from an offset of one, if the above is used).

A way to test the practical use of this change would be to use the [above-mentioned PR](https://github.com/woocommerce/woocommerce-blocks/pull/6812) to see how it could work. Or just pull from `trunk` on [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks).
